tests/Unit/PointwiseFunctions/AnalyticSolutions/Test_Errors.cpp

```diff
@@ -0,0 +1,63 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/ComputeTags.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/System.hpp"
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Errors.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/WrappedGr.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+SPECTRE_TEST_CASE("Unit.AnalyticSolutions.Errors",
+                  "[Unit][PointwiseFunctions]") {
+  static constexpr size_t volume_dim = 1;
+  using system = GeneralizedHarmonic::System<volume_dim>;
+  using solution = GeneralizedHarmonic::Solutions::WrappedGr<
+      gr::Solutions::GaugeWave<volume_dim>>;
+  using solution_tag = Tags::AnalyticSolution<solution>;
+
+  using simple_tags =
+      tmpl::list<domain::Tags::Coordinates<volume_dim, Frame::Inertial>,
+                 Tags::Time, system::variables_tag, solution_tag>;
+  using compute_tags = tmpl::list<
+      evolution::Tags::AnalyticCompute<volume_dim, solution_tag,
+                                       solution::tags<DataVector>>,
+      Tags::DifferenceFromAnalyticCompute<system::variables_tag::tags_list,
+                                          solution::tags<DataVector>>>;
+
+  tnsr::I<DataVector, volume_dim, Frame::Inertial> inertial_coords{
+      {{{1., 2., 3., 4.}}}};
+  constexpr double time = 2.;
+  constexpr double amplitude = 0.1;
+  constexpr double wavelength = 4.0;
+
+  constexpr double constant_numeric_value = 5.0;
+  db::item_type<system::variables_tag> numeric{4, constant_numeric_value};
+
+  const auto box = db::create<db::AddSimpleTags<simple_tags>,
+                              db::AddComputeTags<compute_tags>>(
+      std::move(inertial_coords), time, numeric,
+      solution(amplitude, wavelength));
+
+  tmpl::for_each<system::variables_tag::tags_list>([&box](auto tag_v) {
+    using tensor_tag = typename decltype(tag_v)::type;
+    const auto& analytic = db::get<Tags::Analytic<tensor_tag>>(box);
+    const auto& diff = db::get<Tags::Difference<tensor_tag>>(box);
+    for (size_t i = 0; i < db::item_type<tensor_tag>::size(); ++i) {
```

Comment:
do I need this loop? I thought we could do arithmetic with `Variables`

Reply:
Not when the tags in the variables are different. The analytic Variables has a Tags::Analytic<Tag> for each Tag in the numerical Variables

Reply:
never mind, the loop is over tensors not a Variables...

Reply:
I welcome any suggestions on how to do this with variable arithmetic, but I couldn't figure out how because the Variables have different tags in them (analytic has Tags::Analytic<Tag>, while numeric has Tag). Is there a way to somehow tell Variables which tags go with which, so the arithmetic works?

Reply:
you can't; but why are there two tags lists?  why isn't the AnalyticTagsList constructed from the TagsList?

Reply:
I tried again, and I still don't see how I can do this arithmetic using Variables, because the tags are different for the lhs and rhs

Reply:
I'm not sure I understand your comment @kidder...perhaps we could discuss on a hangout if typing isn't getting the job done? You can't construct the AnalyticTagsList from TagsList via `db::wrap_tags_in<Tags::Analytic, ...>` because AnalyticTagsList might be a superset of TagsList (e.g., if AnalyticCompute puts in a a Variables with everything the analytic solution provides, which might be more than the evolution variables you want to actually compute differences of).

Also, if the analytic solution had the same TagsList as the numeric solution, then there would be an ambiguity which to retrieve when you try to get one from the DataBox.